wordpress-mobile/WordPress-Android
# Dashboard card domain analytics

Fixes #18180 

<img width=320 src="https://user-images.githubusercontent.com/990349/230291552-4864c51c-e672-427f-8b38-c136efe48405.png" />

To test:

1. Launch Jetpack/WordPress app
2. Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
3. Tap **Debug Settings**
4. Find **dashboard_card_domain** under Remote features, enable it, and restart app
5. Verify **domain card** appears as shown above on dashboard
6. Verify `🔵 Tracked: direct_domains_purchase_dashboard_card_shown, Properties: {"position_index":2}` is shown
7. Tap on **domain_card**
8. Verify `🔵 Tracked: direct_domains_purchase_dashboard_card_tapped, Properties: {"position_index":2}`
9. Tap on overflow (three dots) menu
10. Verify `🔵 Tracked: direct_domains_purchase_dashboard_card_menu_tapped, Properties: {"position_index":2}` is shown
11. **Verify** it pops up menu with **Hide this** option
12. Tap on Hide this
13. **Verify**, the card gets hidden 
14. Verify 🔵 Tracked: direct_domains_purchase_dashboard_card_hidden, Properties: {"position_index":2}

>**Note**
>Verify tracks events in Android Studio Logcat or Application Log (Me -> Help -> Logs)


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:
